### ISSUE TITLE
Task `:***:nativeImage` of type `io.micronaut.gradle.graalvm.NativeImageTask`: invocation of 'Task.project' at execution time is unsupported.

### ISSUE THREAD

**Issue by donbeave** [micronaut-projects/micronaut-gradle-plugin#174]
Hello!

I'm playing with Gradle's new feature - Configuration cache: https://docs.gradle.org/current/userguide/configuration_cache.html

Seems almost everything works well, exception tasks related to building or testing `native image`.

**NativeImageTask** almost success, just in the end throwing an exception, but from the log can see that native image was generated successfully.

```
Successfully started process 'command 'native-image''
[application:7715]    classlist:   1,421.41 ms,  0.94 GB
[application:7715]        (cap):   3,394.01 ms,  1.18 GB
[application:7715]        setup:   4,477.04 ms,  1.18 GB
[application:7715]     (clinit):   1,292.32 ms,  6.33 GB
[application:7715]   (typeflow):  16,666.68 ms,  6.33 GB
[application:7715]    (objects):  25,323.44 ms,  6.33 GB
[application:7715]   (features):   2,670.00 ms,  6.33 GB
[application:7715]     analysis:  48,579.90 ms,  6.33 GB
[application:7715]     universe:   1,941.01 ms,  6.33 GB
[application:7715]      (parse):   2,265.34 ms,  6.32 GB
[application:7715]     (inline):   6,391.64 ms,  8.09 GB
[application:7715]    (compile):  14,711.80 ms,  8.30 GB
[application:7715]      compile:  26,771.17 ms,  8.25 GB
[application:7715]        image:   6,216.69 ms,  7.99 GB
[application:7715]        write:   1,642.02 ms,  7.99 GB
[application:7715]      [total]:  91,195.14 ms,  7.99 GB
Native Image written to: /Users/donbeave/Projects/tailrocks/tailrocks-payment/tailrocks-payment-api-app/build/native-image/application
:tailrocks-payment-api-app:nativeImage (Thread[Execution worker for ':',5,main]) completed. Took 1 mins 32.161 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

1 problem was found storing the configuration cache.
- Task `:tailrocks-payment-api-app:nativeImage` of type `io.micronaut.gradle.graalvm.NativeImageTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/6.8.3/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution

See the complete report at file:///Users/donbeave/Projects/tailrocks/tailrocks-payment/build/reports/configuration-cache/8uv9z8agbtyek9jy5rgg8q49f/configuration-cache-report.html
> Invocation of 'Task.project' by task ':tailrocks-payment-api-app:nativeImage' at execution time is unsupported.

* Try:
Run with --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.configurationcache.ConfigurationCacheProblemsException: Configuration cache problems found in this build.

1 problem was found storing the configuration cache.
- Task `:tailrocks-payment-api-app:nativeImage` of type `io.micronaut.gradle.graalvm.NativeImageTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/6.8.3/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution

See the complete report at file:///Users/donbeave/Projects/tailrocks/tailrocks-payment/build/reports/configuration-cache/8uv9z8agbtyek9jy5rgg8q49f/configuration-cache-report.html

...

Caused by: org.gradle.api.InvalidUserCodeException: Invocation of 'Task.project' by task ':tailrocks-payment-api-app:nativeImage' at execution time is unsupported.
        at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onTaskExecutionAccessProblem(ConfigurationCacheProblemsListener.kt:57)
        at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onProjectAccess(ConfigurationCacheProblemsListener.kt:48)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:398)
        at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:380)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:61)
        at org.gradle.internal.event.DefaultListenerManager$EventBroadcast$ListenerDispatch.dispatch(DefaultListenerManager.java:368)
        at org.gradle.internal.event.DefaultListenerManager$EventBroadcast.dispatch(DefaultListenerManager.java:179)
        at org.gradle.internal.event.DefaultListenerManager$EventBroadcast.dispatch(DefaultListenerManager.java:153)
        at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
        at com.sun.proxy.$Proxy112.onProjectAccess(Unknown Source)
        at org.gradle.api.internal.AbstractTask.notifyProjectAccess(AbstractTask.java:978)
        at org.gradle.api.internal.AbstractTask.getProject(AbstractTask.java:220)
        at org.gradle.api.DefaultTask.getProject(DefaultTask.java:57)
        at io.micronaut.gradle.graalvm.NativeImageTask.configure(NativeImageTask.java:105)
        at io.micronaut.gradle.graalvm.NativeImageTask.exec(NativeImageTask.java:95)
```

There is an explanation of what needs to be fixed in [Gradle official documentation](https://docs.gradle.org/6.8.3/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution), I think I can come with PR to fix this issue, just want to know is it ok to do such refactoring? Because it will change the way how this task is configured.

**Comment by graemerocher** [comment]
PRs welcome

**Comment by melix** [comment]
This should be fixed once #228 is merged.
